ec2/sc-ec2-linux-docker-notebook.yaml

```diff
@@ -0,0 +1,513 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: 'Service Catalog: Docker Notebook EC2'
+Metadata:
+  cfn-lint:
+    config:
+      ignore_checks:
+        - E7001
+  AWS::CloudFormation::Interface:
+    ParameterGroups:
+      - Label:
+          default: Linux Instance Configuration
+        Parameters:
+          - EC2InstanceType
+          - NotebookType
+          - VolumeSize
+    ParameterLabels:
+      EC2InstanceType:
+        default: EC2 Instance Type
+      NotebookType:
+        default: Notebook Type
+      VolumeSize:
+        default: Disk Size
+
+Mappings:
+  AccountToImportParams:
+    'Fn::Transform':
+      Name: 'AWS::Include'
+      Parameters:
+        # include is @ https://github.com/Sage-Bionetworks/admincentral-infra/blob/master/templates/cfn-service-catalog-snippets.yaml
+        Location: s3://cfn-snippets-bucket-cloudformationsnippetsbucket-elu83sv8ocdz/scipool-sc-lib-infra/ScAccountToExportMappping.yaml
+  NotebookTypeToInitParams:
+    Jupyter:
+      NoteBookTypeForDocker: jupyter
+      ConfigSet: StartContainersJupyter
+      DockerNotebookContainerName: jupyter-notebook
+      DockerNetworkName: proxy-net
+      DockerImage: quay.io/jupyter/base-notebook:python-3.11
```

Comment:
hmm, would it make sense to allow users to pick a DockerImage instead of a NotebookType?  We can assume the type of notebook the user wants from the docker image that's select, correct?  parameterizing around DockerImage would allow us provide a more specific option.

Reply:
> hmm, would it make sense to allow users to pick a DockerImage instead of a NotebookType?
Yes, but let's refer to each image we allow as a "NotebookType" in this template. E.g., we could add a new type, `RStudioForBMGFki`, which would be a distinct type based on RStudio.

> We can assume the type of notebook the user wants from the docker image that's select, correct?
I would not want to try to guess whether an image is RStudio or Jupyter from the image name. 

> parameterizing around DockerImage would allow us provide a more specific option.
Yes, each "NotebookType" has a `DockerImage` parameter which provides a specific notebook option.

Reply:
> I would not want to try to guess whether an image is RStudio or Jupyter from the image name
why would we need to guess?  either `jupyter` or `rstudio` is in the path of the `DockerImage` parameter, `quay.io/jupyter/base-notebook:python-3.11` and `rocker/rstudio:4.3.2`

Reply:
In the future we may add other RStudio and Jupyter images and the author may not adhere to the naming convention, e.g., they might want to make a Docker repo' called `sagebionetworks/my-project-notebook`.